### PR TITLE
ENG-8817: drain naturally.

### PR DESCRIPTION
**Pull request by akhanzode**

### CODE REVIEW

File: src/frontend/org/voltdb/ImportHandler.java [hunk 1]
```diff
@@ -93,21 +91,25 @@ public void run() {
...
+        try {
+            m_es.submit(new Runnable() {
+
+                @Override
+                public void run() {
+                    try {
+                        //Stop the context first so no more work is submitted.
+                        m_importContext.stop();
+                    } catch (Exception ex) {
+                        ex.printStackTrace();
```
Comment (vtkstef) [inline src/frontend/org/voltdb/ImportHandler.java]: if you are throwing ExecutionException on the `get()` you might as well not ` ex.printStackTrace()` but `Throwables.propagate(ex)`

File: src/frontend/org/voltdb/importclient/kafka/KafkaStreamImporter.java [hunk 2]
```diff
@@ -884,8 +888,12 @@ public void onChange(ImporterChannelAssignment assignment) {
                         HostAndPort hap = m_topicPartitionLeader.get(leaderKey);
                         TopicPartitionFetcher fetcher = new TopicPartitionFetcher(m_brokerList, assignedKey, topic, partition,
                                 hap, m_fetchSize, m_consumerSocketTimeout);
-                        m_fetchers.put(assignedKey.toString(), fetcher);
-                        m_es.submit(fetcher);
+                        try {
```
Comment (vtkstef) [inline src/frontend/org/voltdb/importclient/kafka/KafkaStreamImporter.java]: good change

File: src/frontend/org/voltdb/importer/CSVInvocation.java [hunk 3]
```diff
@@ -42,7 +43,17 @@ public String getProcedure() {
...
     @Override
     public Object[] getParams() throws IOException {
-        return m_parser.parseLine(m_line);
+        Object list[] = m_parser.parseLine(m_line);
+        if (list != null) {
+            for (int i = 0; i < list.length; i++) {
+                if (list[i].equals("NULL")
+                        || list[i].equals(Constants.CSV_NULL)
+                        || list[i].equals(Constants.QUOTED_CSV_NULL)) {
+                    list[i] = null;
```
Comment (vtkstef) [inline src/frontend/org/voltdb/importer/CSVInvocation.java]: better against NPE

```java
              if ("NULL".equals(list[i])
                       || Constants.CSV_NULL.equals(list[i])
                       || Constants.QUOTED_CSV_NULL).equals(list[i])
```

File: src/frontend/org/voltdb/importer/ImportClientResponseAdapter.java [hunk 4]
```diff
@@ -152,15 +137,25 @@ public boolean createTransaction(final String procName, final Procedure catProc,
             executor.submit(new Runnable() {
                 @Override
                 public void run() {
+                    context.hasBackPressure(hasBackPressure());
+                    submitTransaction();
+                }
+                public synchronized boolean submitTransaction() {
```
Comment (vtkstef) [inline src/frontend/org/voltdb/importer/ImportClientResponseAdapter.java]: why synchronized?

File: src/frontend/org/voltdb/importer/ImportClientResponseAdapter.java [hunk 5]
```diff
@@ -152,15 +137,25 @@ public boolean createTransaction(final String procName, final Procedure catProc,
...
                     final long handle = nextHandle();
                     task.setClientHandle(handle);
-                    final ImportCallback cb = new ImportCallback(tcont, procName, proccb, handle);
+                    final ImportCallback cb = new ImportCallback(context, tcont, procName, partition, proccb, handle);
                     m_callbacks.put(handle, cb);
 
                     //Submit the transaction.
-                    VoltDB.instance().getClientInterface().createTransaction(connectionId(), task,
+                    boolean bval = VoltDB.instance().getClientInterface().createTransaction(connectionId(), task,
```
Comment (vtkstef) [inline src/frontend/org/voltdb/importer/ImportClientResponseAdapter.java]: wrap `VoltDB.instance().getClientInterface()` inside a a `getClientInterface()` method to better support mocking

**Comment by akhanzode** [comment]
squash merged. r=vtkstef

**Pull request closed without merging**